### PR TITLE
streamingccl: add WITH RETENTION option to CREATE TENANT

### PR DESCRIPTION
**Pull request by stevendanna**
This adds an WITH RETENTION = 'duration' option to CREATE TENANT to allow the user to set the retention window at replication stream creation time.

A future commit will support updating this option via SET as well.

Informs: #93441

Release note: None

### CODE REVIEW

File: pkg/sql/parser/sql.y [hunk 1]
```diff
@@ -4146,16 +4150,52 @@ create_tenant_stmt:
...
   }
 | CREATE TENANT error // SHOW HELP: CREATE TENANT
 
+// Optional tenant replication options.
+opt_with_tenant_replication_options:
+  WITH tenant_replication_options_list
+  {
+    $$.val = $2.tenantReplicationOptions()
+  }
+| WITH OPTIONS '(' tenant_replication_options_list ')'
```
Comment (lidorcarmel) [inline pkg/sql/parser/sql.y:4170]: so we can also do `CREATE TENANT .. FROM REPLICATION OF .. ON .. WITH OPTIONS (RETENTION = 36h);` right? can you add a test for that syntax? or maybe we can wait for the data driven tests for that?

btw is that standard to support both `WITH X = y` and `WITH OPTIONS (X = y)`?

**Review by lidorcarmel — APPROVED** [review]
looks great, just some nagging questions..

File: pkg/sql/parser/sql.y (hunk 1 above)
Comment (stevendanna) [reply pkg/sql/parser/sql.y:4170]: Added a parser test.

The `WITH OPTIONS (X =y)` is also supported by BACKUP and RESTORE so I supported it here for consistency.

**Comment by stevendanna** [comment]
bors r=lidorcarmel

**Pull request merged**